File: backend/app/schemas/catalog.py
"""仿真目录响应Schema"""

from __future__ import annotations

from pydantic import BaseModel


class OriginSchema(BaseModel):
    origin_id: str
    label: str
    lane_ids: list[str]


class LaneSchema(BaseModel):
    lane_id: str
    edge_id: str
    lane_index: int
    role: str
    approach: str | None
    approach_label: str | None
    length: float
    max_speed: float


class IntersectionSchema(BaseModel):
    intersection_id: str
    longitude: float | None
    latitude: float | None
    periods: list[str]
    origins: list[OriginSchema]
    lanes: list[LaneSchema]


class FlowMultiplierRangeSchema(BaseModel):
    min: float
    max: float


class CatalogResponse(BaseModel):
    intersections: list[IntersectionSchema]
    event_types: list[str]
    control_modes: list[str]
    flow_multiplier: FlowMultiplierRangeSchema

File: frontend/src/types/catalog.ts
export interface CatalogOrigin {
  origin_id: string
  label: string
  lane_ids: string[]
}

export interface CatalogLane {
  lane_id: string
  edge_id: string
  lane_index: number
  role: string
  approach: string | null
  approach_label: string | null
  length: number
  max_speed: number
}

export interface CatalogIntersection {
  intersection_id: string
  longitude: number | null
  latitude: number | null
  periods: string[]
  origins: CatalogOrigin[]
  lanes: CatalogLane[]
}

export interface CatalogScenarioPreset {
  preset_id: string
  label: string
  intersection_ids: string[]
  map_template: string
}

export interface CatalogResponse {
  intersections: CatalogIntersection[]
  scenario_presets: CatalogScenarioPreset[]
  event_types: string[]
  control_modes: string[]
  playback_speeds: number[]
}
